docs/EKS-cni-support.md

```diff
@@ -0,0 +1,42 @@
+# CNI monitoring special configuration updates for EKS
+
+AWS EKS uses [CNI](https://github.com/aws/amazon-vpc-cni-k8s) networking plugin for pod networking in Kubernetes using Elastic Network Interfaces on AWS
+
+One fatal issue that can occur is that you run out of IP addresses in your eks cluster. (Generally happens due to error configs where pods keep scheduling).
+
+You can monitor the `awscni` using kube-promethus with : 
+[embedmd]:# (../examples/eks-cni-example.jsonnet)
+```jsonnet
+local kp =  (import 'kube-prometheus/kube-prometheus.libsonnet') +
+            (import 'kube-prometheus/kube-prometheus-eks.libsonnet') + {
+  _config+:: {
+    namespace: 'monitoring',
+  },
+  prometheusRules+:: {
+    groups+: [
+      {
+        name: 'example-group',
+        rules: [
+          {
+            record: 'aws_eks_available_ip',
```

Comment:
What speaks against having all eks clusters benefit from this? :)

As in why just put it in the example and not part of the patch?

Reply:
@brancz sorry for being super late here.
a3ab6bd49b39158222eef7309e1a28fc17f1dc07 this adds the rule in the patch itself.
I am figuring out other metrics as well.. will add them in future (with more alerts, etc.)

Reply:
Awesome! Once CI passes, I'm super happy to merge this! :)

Reply:
I am figuring out why it is failing.. it started to happen after I rebased my branch with master.. 🙄